File: Selenium/Activities/Activity4.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;

public class Activity4 {
    public static void main(String[] args) {
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        driver.get("https://v1.training-support.net/selenium/target-practice");
        System.out.println("Title of webpage:"+driver.getTitle());
        System.out.println(driver.findElement(By.xpath("//h3")).getText());
        System.out.println(driver.findElement(By.xpath("//h5")).getCssValue("color"));
        System.out.println(driver.findElement(By.xpath("//button[text()='Violet']")).getAttribute("class"));
        System.out.println(driver.findElement(By.xpath("//button[text()='Grey']")).getText());
        driver.close();
    }
}

File: Selenium/Activities/Activity4.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Navigate to the URL
    driver.get("https://v1.training-support.net/selenium/target-practice")
    print("Title of webpage:" + driver.title)
    print(driver.find_element(By.XPATH, "//h3").text)
    print(driver.find_element(By.XPATH, "//h5").value_of_css_property("color"))
    print(driver.find_element(By.XPATH,"//button[text()='Violet']").get_attribute("class"))
    print(driver.find_element(By.XPATH,"//button[text()='Grey']").text)
    driver.close()


